### ISSUE TITLE
How to flow your appsettings from startup.cs into your XBot.xs file?

### ISSUE THREAD

**Issue by jeffrey-opdam** [Microsoft/botbuilder-dotnet#592]
How do you flow your appsettings, from startup.cs, where you retrieved them, into your bot?

What would the pattern look like?

**Comment by drub0y** [comment]
TL;DR; - if you're familiar with how options are configured in ASP.NET Core, just follow the same approach and then request an `IOptions<YourOptions>` to your bot's constructor and it will be injected just like it would for a controller.

The integration library for ASP.NET Core leverages the same dependency injection container as the rest of ASP.NET Core to resolve the configured `IBot`. This means that you can inject any dependency you need into your bot just like you would for any MVC controller.

This starts with loading your settings via the standard configuration API like so via a pretty much boilerplate ASP.NET `Startup.cs`:

#### Startup.cs

```CSharp
public class Startup
    {
        public Startup(IHostingEnvironment env)
        {
            var builder = new ConfigurationBuilder()
                .SetBasePath(env.ContentRootPath)
                .AddJsonFile("appsettings.json", optional: true, reloadOnChange: true)
                .AddJsonFile($"appsettings.{env.EnvironmentName}.json", optional: true)
                .AddEnvironmentVariables();

            Configuration = builder.Build();
        }

        public IConfigurationRoot Configuration { get; private set; }

        public void ConfigureServices(IServiceCollection services)
        {
            services.Configure<MyOptions>(Configuration.GetSection("MyOptions"));
            services.AddBot<MyBot>(options =>
            {
                options.CredentialProvider = new SimpleCredentialProvider();
            });
        }

        public void Configure(IApplicationBuilder app, IHostingEnvironment env)
        {
            if (env.IsDevelopment())
            {
                app.UseDeveloperExceptionPage();
            }

            app.UseBotFramework();

            app.Run(async (context) =>
            {
                await context.Response.WriteAsync("Welcome, this sample shows how to configure your bot with options!");
            });
        }
    }
```

Now I happened to define `MyOptions` like this:

#### MyOptions.cs

```CSharp
    public class MyOptions
    {
        public int MyFirstOption { get; set; }
        public string MySecondOption { get; set; }
    }
```
Then all that's left to do is ask for those options to be injected into your bot by adding a parameter to your bot's constructor. My bot then looks like this:

#### MyBot.cs

```CSharp
public class MyBot : IBot
    {
        private readonly IOptions<MyOptions> _options;

        public MyBot(IOptions<MyOptions> options)
        {
            _options = options ?? throw new System.ArgumentNullException(nameof(options));
        }

        public async Task OnTurn(ITurnContext turnContext)
        {
            await turnContext.SendActivity($"Hey, check out my options:\r\n\r\n\r\n\r\nMyFirstOption={_options.Value.MyFirstOption}\r\n\r\nMySecondOption={_options.Value.MySecondOption}");
        }
    }
```

If this doesn't answer your question or you have any more just let me know!
